webui/src/pages/auth/login.tsx

```diff
@@ -53,8 +95,8 @@ const LoginForm = ({loginConfig}: {loginConfig: LoginConfig}) => {
                             const username = formData.get('username');
                             const password = formData.get('password');
                             await auth.login(username, password);
-                            router.push(next || '/');
-                            navigate(0);
+                            setAuthStatus(AUTH_STATUS.AUTHENTICATED);
```

Comment:
I suspect there might be a race condition here, between `setAuthStatus` and `router.navigate`.
How about making sure that the `router.navigate` will happen _after_ the `setAuthStatus`?